src/uct/ib/rc/accel/rc_mlx5_common.c

```diff
@@ -104,30 +112,37 @@ unsigned uct_rc_mlx5_iface_srq_post_recv(uct_rc_iface_t *iface, uct_ib_mlx5_srq_
             srq->free_idx  = next_index;
         }
 
-        if (seg->srq.desc == NULL) {
-            UCT_TL_IFACE_GET_RX_DESC(&iface->super.super, &iface->rx.mp,
-                                     desc, break);
+        for (i = 0; i < iface->tm.mp.num_strides; ++i) {
+            if (seg->srq.ptr_mask & UCS_BIT(i)) {
```

Comment:
Would need to do something like `ucs_for_each_bit(i, ~seg->srq.ptr_mask & UCS_MASK(iface->tm.mp.num_strides))`
Or change `ptr_mask` in a way that `1` bit would mean _taken_ desc, which needs to be reposted
Also not sure it will be noticeably faster  (considering that posting is not a fast path)

Reply:
ok, but please note that we scan only `num_strides` bits ( 1 or 8 in most cases )

Reply:
for 8 case i think it's worth it